dd-smoke-tests/java9-modules/build.gradle

```
@@ -13,17 +13,16 @@ jar {
 sourceCompatibility = JavaVersion.VERSION_1_8
 targetCompatibility = JavaVersion.VERSION_1_8
 
-[JavaCompile, GroovyCompile].each {
-  tasks.withType(it).configureEach { AbstractCompile ac ->
-    setJavaVersion(ac, 11)
-    // read up classpath lazily to avoid early locking of gradle settings
-    ac.options.compilerArgumentProviders.add(new CommandLineArgumentProvider() {
-        @Override
-        Iterable<String> asArguments() {
-          { return ['--module-path', ac.classpath.asPath]}
-        }
-      })
-  }
+tasks.withType(JavaCompile).configureEach {
+  setJavaVersion(it, 11)
+  def ac = it
+  // read up classpath lazily to avoid early locking of gradle settings
+  it.options.compilerArgumentProviders.add(new CommandLineArgumentProvider() {
+      @Override
+      Iterable<String> asArguments() {
+        { return ['--module-path', ac.classpath.asPath]}
+      }
+    })
```

Not a Gradle expert but would that work?
```suggestion
tasks.withType(JavaCompile).configureEach { JavaCompile jc ->
  setJavaVersion(jc, 11)
  // read up classpath lazily to avoid early locking of gradle settings
  it.options.compilerArgumentProviders.add(new CommandLineArgumentProvider() {
      @Override
      Iterable<String> asArguments() {
        { return ['--module-path', jc.classpath.asPath]}
      }
    })
```